lib/rules/no-unused-prop-types.js

```diff
@@ -992,7 +928,8 @@ module.exports = {
         // let {firstname} = props
         const statelessDestructuring = destructuring && isPropAttributeName(node) && (
           utils.getParentStatelessComponent() ||
-          isInLifeCycleMethod(node)
+          astUtil.inConstructor(context) ||
```

Comment:
also here, which test case exposes the need for this check?

Reply:
Same as above: Note that the old isInLifeCycleMethod function already was checking whether we are in constructor: node.kind === 'constructor'. Now it is split into inConstructor and inLifecycleMethod so the code is the same between this rule and prop-types.

However, this case seems to be untested. 😞 In master we can remove this whole block and the tests still pass.

```
&& ( utils.getParentStatelessComponent() || isInLifeCycleMethod(node))
```